app/main.py

```diff
@@ -1,13 +1,40 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error["line_number"], "column": error["column_number"],
+        "message": error["text"], "name": error["code"], "source": "flake8"
+    }
```

Comment:
The `format_linter_error` function is correctly implemented, but its logic is duplicated in the `format_single_linter_file` and `format_linter_report` functions. Consider creating a helper function to avoid repetition.